src/models/injector/GenericClassDecorator.ts

```diff
@@ -0,0 +1,9 @@
+/**
+ * Type for what object is instances of
+ */
+export type Type<T> = new(...args: any[]) => T;
```

Comment:
very smart, nice utility 👍

Reply:
This is actually used in all of our repos for DVSA. Nothing special from my end :)